### PR TITLE
reuse `Tag` in `SelectTag`

### PR DESCRIPTION
**Pull request by mayank99**
## Changes

This PR takes a small step towards #907.

- **React**: Updated `SelectTag` component to use `Tag` internally and add extra classes on top of it.
- **CSS**: Removed the use of mixins in `select-tag`. Both `tag` and `select-tag` classes need to be present now.
  - Also replaced the `select-tag-button-icon` class with a nested `svg` selector.

There are two important outcomes to these changes:
1. We are able to make use of the "remove button" code from `Tag` (which also handles some edge cases; see #1824)
2. We ship less CSS (mixins create unnecessary duplication).

## Testing

All existing tests passing. No visual changes.

## Docs

Added `major` CSS changeset for DOM structure changes.

Doesn't affect usage or behavior for React package consumers.

### CODE REVIEW

File: packages/itwinui-css/src/select/select.scss [hunk 1]
```diff
@@ -153,18 +147,14 @@
   @include iui-select-tag-container;
 }
 
-.iui-select-tag {
+.iui-select-tag:is(.iui-tag) {
```
[the remark this answers is not in the document]
Comment (mayank99) [reply packages/itwinui-css/src/select/select.scss:150]: It's mainly for increasing specificity, since `select.scss` is ordered before `tag.scss` and we want these styles to take preference.

And yes, we do want both classes present together. Think of `:is(.iui-tag)` as replacing `@include iui-tag`.

**Review by smmr-dn — APPROVED** [review]
